doc/adapters.md

```diff
@@ -0,0 +1,42 @@
+# WASM Runtime Adapters
+
+The test runner is designed to support different types of WebAssembly runtimes. Because the command line interface for WASM runtime is not standardized, every runtime can arbitrarily define how parameters should be passed to the runtime.
+
+WASI test runner is designed to support as many WASM runtimes as possible, therefore runtime peculiarities aren't hardcoded in the runner.
+
+In order to integrate WASM runtime with a test runner, the user has to provide a `runtime adapter`. It's an executable file that takes command-line arguments and translates them to a runtime call.
```

Comment:
do you mave any recommendations how to give misc configurations like the followings?
* runtime executable path
* aot compiler path
* runtime specific options to the above executables

Reply:
i agree 1 is fine as far as our scope is simple wasi commands.

Reply:
Created a ticket for that: https://github.com/WebAssembly/wasi-testsuite/issues/19